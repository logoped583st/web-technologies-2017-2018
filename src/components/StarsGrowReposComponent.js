import React from 'react';
import { bindActionCreators } from 'redux';
import { fetchStarsReposRequest } from '../actions/actions'
import { fetchGrowReposRequest } from '../actions/actions'

import { connect } from 'react-redux';

class StarsGrowReposComponent extends React.Component {

    constructor(props) {
        super(props);
        this.state = { tabIndex: 0 };
        if (props.request === 'TOP') {
            this.props.fetchStarsReposRequest();
        } else if (props.request === 'GROW') {
            this.props.fetchGrowReposRequest();
        }
    }




    render() {
        return (

            (this.props.repos.items !== undefined && this.props.repos.items.length !== 0) && <div className="REPS">

                <h2>Repositories</h2>
                {this.props.repos.items.map(function (item) {
                    return (<div className="orgs_adds">
                        <h2>{item.name}</h2>
                        <h3> {item.html_url}   <a>{item.language}</a></h3>
                    </div>)
                })}
            </div>

        )
    }
}

const getState = (state) => {
    console.log(state)
    return {
        repos: state.reducerStarsGrowRepos
    };
};

const mapDispatchToProps = dispatch => bindActionCreators({
    fetchStarsReposRequest,
    fetchGrowReposRequest,
}, dispatch);


export default connect(getState, mapDispatchToProps)(StarsGrowReposComponent);