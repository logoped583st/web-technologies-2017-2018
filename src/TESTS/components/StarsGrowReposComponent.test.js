import React from 'react';
import StarsGrowReposComponent from '../../components/StarsGrowReposComponent';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('StarsGrowReposComponent component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {
        reducerStarsGrowRepos:
            [
                {
                    name: 'asdasd',
                    html_url: 'http://KAAAAKAA/',
                    language: 'Java',
                    private: false,
                },
            ]

    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('starsGrowReposComponent component rendered succesfully', () => {
        const starsGrowReposComponent = mount(
            <StarsGrowReposComponent store={store} />
        );
        expect(mountToJson(starsGrowReposComponent)).toMatchSnapshot();
    });
    
    it('starsGrowReposComponent component rendered succesfully', () => {
       
        const fakeInitialState = {
           
            reducerStarsGrowRepos: undefined

                
            
        }
        const fakeStore = mockStore(fakeInitialState);
        const userRepositories = mount(
            
                <StarsGrowReposComponent store={fakeStore} />
            
        );
        expect(userRepositories.prop('reducerStarsGrowRepos')).toBe(undefined);
        expect(mountToJson(userRepositories)).toMatchSnapshot();
    });

 
});