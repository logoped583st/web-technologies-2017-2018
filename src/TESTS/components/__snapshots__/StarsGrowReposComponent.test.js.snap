// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarsGrowReposComponent component test starsGrowReposComponent component rendered succesfully 1`] = `
<Connect(StarsGrowReposComponent)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <StarsGrowReposComponent
    fetchGrowReposRequest={[Function]}
    fetchStarsReposRequest={[Function]}
    repos={
      Array [
        Object {
          "html_url": "http://KAAAAKAA/",
          "language": "Java",
          "name": "asdasd",
          "private": false,
        },
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />
</Connect(StarsGrowReposComponent)>
`;

exports[`StarsGrowReposComponent component test starsGrowReposComponent component rendered succesfully 2`] = `
<Connect(StarsGrowReposComponent)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <StarsGrowReposComponent
    fetchGrowReposRequest={[Function]}
    fetchStarsReposRequest={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />
</Connect(StarsGrowReposComponent)>
`;
